spyder/config/user.py

```diff
@@ -447,14 +584,376 @@ def set(self, section, option, value, verbose=False, save=True):
+
+        if version is None:
+            fpath = self.get_config_fpath()
+        elif check_version(version, '52.0.0', '<'):
+            fpath = osp.join(get_conf_path(), 'spyder.ini')
+        elif check_version(version, '2.4.0', '<'):
+            # TODO: Should this still be supported?
+            pass
+
+        return fpath
+
+    def get_backup_fpath_from_version(self, version=None, old_version=None):
+        """
+        Override method.
+
+        Make a backup of the configuration file.
+        """
+        if old_version and check_version(old_version, '52.0.0', '<'):
+            name = 'spyder.ini'
+            fpath = os.path.join(get_conf_path(), name)
+            if version is None:
+                backup_fpath = "{}{}".format(fpath, self._backup_suffix)
+            else:
+                backup_fpath = "{}-{}{}".format(fpath, version,
+                                                self._backup_suffix)
+        else:
+            super_class = super(SpyderUserConfig, self)
+            backup_fpath = super_class.get_backup_fpath_from_version(
+                version, old_version)
+
+        return backup_fpath
+
+    # TODO:
+    def get_defaults_path_name_from_version(self, old_version=None):
+        """
+        Override method.
+
+        Get defaults location based on version.
+        """
+        if old_version:
+            if check_version(old_version, '3.0.0', '<='):
+                name = '{}-{}'.format('defaults', old_version)
+                path = self._module_source_path
+            elif check_version(old_version, '52.0.0', '<'):
+                name = '{}-{}'.format(self._defaults_name_prefix, old_version)
+                path = osp.join(get_conf_path(), 'defaults')
+        else:
+            super_class = super(SpyderUserConfig, self)
+            path, name = super_class.get_defaults_path_name_from_version()
+
+        return path, name
+
+    def apply_configuration_patches(self, old_version=None):
+        """
+        Override method.
+
+        Apply any patch to configuration values on version changes.
+        """
+        # TODO: is this check still needed?
+        if old_version and check_version(old_version, '2.4.0', '<'):
+            self.reset_to_defaults(save=False)
+        else:
+            # TODO: Remove comment if tests pass ok
+            # self._update_defaults(defaults, old_version)
+            self._update_defaults(self.defaults, old_version)
+
+        if old_version and check_version(old_version, '44.1.0', '<'):
+            run_lines = to_text_string(self.get('ipython_console',
+                                                'startup/run_lines'))
+            if run_lines is not NoDefault:
+                run_lines = run_lines.replace(',', '; ')
+                self.set('ipython_console', 'startup/run_lines', run_lines)
+
+
+class MultiUserConfig(object):
+    """
+    Multi user config class based emulating the basic interface to UserConfig.

```

Comment:
Please change this to

    Multiuser config class which emulates the basic UserConfig interface.

Reply:
Done